x-pack/plugins/infra/public/pages/metrics/hosts/components/hosts_table.tsx

```diff
@@ -5,549 +5,23 @@
-            width: 121.11666666666667,
-          },
-          {
-            columnId: '3eca2307-228e-4842-a023-57e15c8c364d',
-            isTransposed: false,
-          },
-          {
-            columnId: '0a9bd0fa-9966-489b-8c95-70997a7aad4c',
-            isTransposed: false,
-          },
-          {
-            columnId: 'fe5a4d7d-6f48-45ab-974c-96bc864ac36f',
-            isTransposed: false,
-          },
-          {
-            isTransposed: false,
-            columnId: '02e9d54c-bbe0-42dc-839c-55080a29838d',
-          },
-          {
-            isTransposed: false,
-            columnId: '7802ef93-622c-44df-94fa-03eec01bb7b6',
-          },
-        ],
-        paging: {
-          size: 10,
-          enabled: true,
-        },
-        headerRowHeight: 'custom',
-        headerRowHeightLines: 2,
-        rowHeight: 'single',
-        rowHeightLines: 1,
-      },
-      filters: [],
-      query,
-    },
-  } as TypedLensByValueInput['attributes']);
-
+import React from 'react';
+import { EuiBasicTable } from '@elastic/eui';
+import { SnapshotNode } from '../../../../../common/http_api';
+import { HostsTableColumns } from './hosts_table_columns';
+import { useHostTable } from '../hooks/use_host_table';
 interface Props {
-  dataView: DataView;
-  timeRange: TimeRange;
-  query: Query;
-  searchSessionId: string;
-  onRefetch: () => void;
-  onLoading: (isLoading: boolean) => void;
-  isLensLoading: boolean;
+  nodes: SnapshotNode[];
 }
-export const HostsTable: React.FunctionComponent<Props> = ({
-  dataView,
-  timeRange,
-  query,
-  searchSessionId,
-  onRefetch,
-  onLoading,
-  isLensLoading,
-}) => {
-  const {
-    services: { lens },
-  } = useKibana<InfraClientStartDeps>();
-  const LensComponent = lens?.EmbeddableComponent;
-  const [noData, setNoData] = useState(false);
 
-  if (noData && !isLensLoading) {
-    return (
-      <NoData
-        titleText={i18n.translate('xpack.infra.metrics.emptyViewTitle', {
-          defaultMessage: 'There is no data to display.',
-        })}
-        bodyText={i18n.translate('xpack.infra.metrics.emptyViewDescription', {
-          defaultMessage: 'Try adjusting your time or filter.',
-        })}
-        refetchText={i18n.translate('xpack.infra.metrics.refetchButtonLabel', {
-          defaultMessage: 'Check for new data',
-        })}
-        onRefetch={onRefetch}
-        testString="metricsEmptyViewState"
-      />
-    );
-  }
+export const HostsTable: React.FunctionComponent<Props> = ({ nodes }) => {
+  const items = useHostTable(nodes);
+
   return (
-    <LensComponent
-      id="hostsView"
-      timeRange={timeRange}
-      attributes={getLensHostsTable(dataView, query)}
-      searchSessionId={searchSessionId}
-      onLoad={(isLoading, adapters) => {
-        if (!isLoading && adapters?.tables) {
-          setNoData(adapters?.tables.tables.default?.rows.length === 0);
-          onLoading(false);
-        }
-      }}
+    <EuiBasicTable
```

Comment:
Should we also add the sorting of the table or we will do it in another issue? I couldn't find anything mentioned in the issue and I am not sure if we had it out of the box with the Lens table 🤔

Reply:
Issue added: https://github.com/elastic/kibana/issues/143510